src/Serialization/Fields.jl

```diff
@@ -268,6 +268,79 @@ function load_internal_with_parent(s::DeserializerState,
     return parent(num, den)
 end
 
+################################################################################
+# RationalFunctionField
+
+@registerSerializationType(AbstractAlgebra.Generic.RationalFunctionField)
+
+function save_internal(s::SerializerState,
+                       RF::AbstractAlgebra.Generic.RationalFunctionField)
+    return Dict(
+        :base_ring => save_type_dispatch(s, base_ring(RF)),
+        :symbols => save_type_dispatch(s, symbols(RF))
+    )
+end
+
+function load_internal(s::DeserializerState,
+                       ::Type{<: AbstractAlgebra.Generic.RationalFunctionField},
+                       dict::Dict)
+    R = load_unknown_type(s, dict[:base_ring])
+    symbols = load_unknown_type(s, dict[:symbols])
+
+    return RationalFunctionField(R, symbols, cached=false)[1]
+end
+
+#elements
+@registerSerializationType(AbstractAlgebra.Generic.Rat)
+
+function save_internal(s::SerializerState, f::AbstractAlgebra.Generic.Rat)
+    frac_elem_parent = save_type_dispatch(s, parent(denominator(f)))
+    return Dict(
+        :parent => save_type_dispatch(s, parent(f)),
+        :frac_elem_parent => frac_elem_parent,
+        :den => save_type_dispatch(s, denominator(f)),
+        :num => save_type_dispatch(s, numerator(f))
+    )
+end
+
+function load_internal(s::DeserializerState,
+                       ::Type{<: AbstractAlgebra.Generic.Rat},
+                       dict::Dict)
+    _ = load_unknown_type(s, dict[:frac_elem_parent])
+    R = load_type_dispatch(s, AbstractAlgebra.Generic.RationalFunctionField, dict[:parent])
+    num = load_unknown_type(s, dict[:num])
+    den = load_unknown_type(s, dict[:den])
+
+    if num isa PolyElem
+        num = evaluate(num, gen(R))
+        den = evaluate(den, gen(R))
+    else
+        num = evaluate(num, gens(R))
+        den = evaluate(den, gens(R))
+    end
+
+    return num // den
```

Comment:
```suggestion
return R(num, den)
```

I am not sure this evaluation and // is even necessary. AbstractAlgebra/src/generic/RationalFunctionField.jl has
```
function (a::RationalFunctionField{T, U})(n::U, d::U) where {T <: FieldElement, U <: Union{PolyElem{T}, MPolyElem{T}}}
```

Reply:
so i originally tried this but i ran into this
```
julia> Qt, t = RationalFunctionField(QQ, :t)
(Rational function field over Rational Field, t)

julia> save("/homes/combi/vecchia/local/misc-data/test-file.json", t)
1852

julia> loaded = load("/homes/combi/vecchia/local/misc-data/test-file.json")
ERROR: Unable to coerce rational function
Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:33
  [2] RationalFunctionField
    @ ~/local/.julia/packages/AbstractAlgebra/ElVzc/src/generic/RationalFunctionField.jl:533 [inlined]
  [3] (::AbstractAlgebra.Generic.RationalFunctionField{fmpq, fmpq_poly})(n::fmpq_poly, d::fmpq_poly)
    @ AbstractAlgebra.Generic ~/local/.julia/packages/AbstractAlgebra/ElVzc/src/generic/RationalFunctionField.jl:552
  [4] load_internal(s::Oscar.DeserializerState, #unused#::Type{AbstractAlgebra.Generic.Rat}, dict::Dict{Symbol, Any})
    @ Oscar /home/datastore/vecchia/repositories/Oscar.jl/src/Serialization/Fields.jl:322
  [5] load_type_dispatch(s::Oscar.DeserializerState, ::Type{AbstractAlgebra.Generic.Rat}, dict::Dict{Symbol, Any}; parent::Nothing)
    @ Oscar /home/datastore/vecchia/repositories/Oscar.jl/src/Serialization/main.jl:152
  [6] load_unknown_type(s::Oscar.DeserializerState, dict::Dict{Symbol, Any}; parent::Nothing,
check_namespace::Bool)
    @ Oscar /home/datastore/vecchia/repositories/Oscar.jl/src/Serialization/main.jl:181
  [7] load(io::IOStream; parent::Nothing)
    @ Oscar /home/datastore/vecchia/repositories/Oscar.jl/src/Serialization/main.jl:264
  [8] #3135
    @ /home/datastore/vecchia/repositories/Oscar.jl/src/Serialization/main.jl:269 [inlined]
  [9] open(f::Oscar.var"#3135#3136"{Nothing}, args::String; kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Base ./io.jl:330
 [10] open
    @ ./io.jl:328 [inlined]
 [11] #load#3134
    @ /home/datastore/vecchia/repositories/Oscar.jl/src/Serialization/main.jl:268 [inlined]
 [12] load(filename::String)
    @ Oscar /home/datastore/vecchia/repositories/Oscar.jl/src/Serialization/main.jl:268
 [13] top-level scope
    @ REPL[5]:1
```

Reply:
Ok.  I will have a look.

Reply:
The issue is that `num` and `den` have the wrong parent. They both have the same parent, but this common parent is different from what `R` expects. I debugged it with
```
    @show parent(num)
    @show parent(den)
    @show base_ring(AbstractAlgebra.Generic.fraction_field(R))

    @show parent(num) == parent(den)      # true
    @show parent(num) == base_ring(AbstractAlgebra.Generic.fraction_field(R))    # false

    return R(num, den)
```
Is there a way you can deserialize `num` and `den` into `base_ring(AbstractAlgebra.Generic.fraction_field(R))`.

By the way, it should be possible to know the type of `num` and `den`. If the rat is a `AbstractAlgebra.Generic.Rat{T, U}`, then `num` and `den` are `U`s.